pkg/gateway/gateway.go

```diff
@@ -618,11 +618,19 @@ func (n *jfsObjects) PutObject(ctx context.Context, bucket string, object string
 	}
 
 	p := n.path(bucket, object)
-	if strings.HasSuffix(object, sep) && r.Size() == 0 {
+	if strings.HasSuffix(object, sep) {
 		if err = n.mkdirAll(ctx, p, os.FileMode(0755)); err != nil {
 			err = jfsToObjectErr(ctx, err, bucket, object)
 			return
 		}
+		if r.Size() > 0 {
```

Comment:
i think if r.Size()>0, we needn't create dir. this condition should before mkdirAll

Reply:
We see this behavior in MinIO, so we'd like to keep this side-affected behavior.